src/api/interceptors.rs

```diff
@@ -187,8 +207,14 @@ impl Interceptor for ServiceAccountInterceptor {
                 .join()
                 .map_err(|_| Status::internal("could not fetch token"))??;
 
-            self.token = Some(token.clone());
-            self.token_expiry = Some(time::OffsetDateTime::now_utc() + time::Duration::minutes(59));
```

Comment:
Isn"t this a bug? Shouldn't the token expiry be used from the endpoint?

Reply:
If I recall correctly this is the "local" token that authorizes the application against ZITADEL, not the cached token from the user. This token is used, when no client secret is used and the application authenticates itself with an rsa key.

Reply:
but where did the hardcoded expiry come from?

Reply:
But isn't this also an access token? This should also have a dedicated expiry, which could change

Reply:
@buehler I will investigate this outside of this PR. So the proposed change should not block it.